client/post-editor/media-modal/image-editor/image-editor-canvas.jsx

```diff
@@ -74,26 +81,49 @@ const MediaModalImageEditorCanvas = React.createClass( {
 	},
 
 	toBlob( callback ) {
-		const canvas = ReactDom.findDOMNode( this.refs.canvas );
-
-		MediaUtils.canvasToBlob( canvas, callback, this.props.mimeType, 1 );
+		const canvas = ReactDom.findDOMNode( this.refs.canvas ),
+			context = canvas.getContext( '2d' ),
+			croppedLeft = this.props.cropLeftRatio * this.image.width,
+			croppedTop = this.props.cropTopRatio * this.image.height,
+			croppedWidth = this.props.cropWidthRatio * this.image.width,
+			croppedHeight = this.props.cropHeightRatio * this.image.height,
+			imageData = context.getImageData(
+				croppedLeft,
+				croppedTop,
+				croppedWidth,
+				croppedHeight ),
+			newCanvas = document.createElement( 'canvas' );
+
+		newCanvas.width = croppedWidth;
+		newCanvas.height = croppedHeight;
+
+		const newContext = newCanvas.getContext( '2d' );
+		newContext.putImageData( imageData, 0, 0 );
+
+		MediaUtils.canvasToBlob( newCanvas, callback, this.props.mimeType, 1 );
 	},
 
 	drawImage() {
 		if ( ! this.image ) {
 			return;
 		}
 
-		if ( this.state.canvasWidth !== this.image.width ||
-			this.state.canvasHeight !== this.image.height ) {
-			this.setState( {
-				canvasWidth: this.image.width,
-				canvasHeight: this.image.height
-			} );
-		}
-
 		const canvas = ReactDom.findDOMNode( this.refs.canvas ),
-			context = canvas.getContext( '2d' );
+			imageWidth = this.image.width,
+			imageHeight = this.image.height,
+			rotated = this.props.degrees % 180 !== 0;
+
+		//make sure the canvas draw area is the same size as the image
+		canvas.width = rotated ? imageHeight : imageWidth;
+		canvas.height = rotated ? imageWidth : imageHeight;
+
+		this.props.setImageEditorCropBounds(
+			canvas.offsetTop - canvas.offsetHeight / 2,
```

Comment:
Looking more closely at the code this only updates on button press. So caching the bounding box information is optional here.